application/src/main/java/org/togetherjava/tjbot/commands/reminder/ReminderCommand.java

```diff
@@ -125,40 +159,84 @@ private void handleCreateCommand(SlashCommandInteractionEvent event) {
+                getPendingReminders(event.getGuild(), event.getUser());
 
-        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
-            .setColor(RemindRoutine.AMBIENT_COLOR);
+        event.reply(createListMessage(pendingReminders, 1)).setEphemeral(true).queue();
+    }
 
-        long guildId = event.getGuild().getIdLong();
-        long userId = event.getUser().getIdLong();
+    private Result<PendingRemindersRecord> getPendingReminders(Guild guild, User user) {
+        return database.read(context -> context.selectFrom(PENDING_REMINDERS)
+            .where(PENDING_REMINDERS.GUILD_ID.eq(guild.getIdLong())
+                .and(PENDING_REMINDERS.AUTHOR_ID.eq(user.getIdLong())))
+            .orderBy(PENDING_REMINDERS.REMIND_AT.asc())
+            .fetch());
+    }
 
-        Result<PendingRemindersRecord> pendingReminders = getReminders(guildId, userId);
+    private MessageCreateData createListMessage(Result<PendingRemindersRecord> pendingReminders,
+            int pageToShow) {
+        int totalPages = Math.ceilDiv(pendingReminders.size(), REMINDERS_PER_PAGE);
+
+        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
+            .setColor(RemindRoutine.AMBIENT_COLOR);
+        MessageCreateBuilder listMessage = new MessageCreateBuilder();
 
         if (pendingReminders.isEmpty()) {
             remindersEmbed.setDescription("No pending reminders");
         } else {
-            pendingReminders.forEach(reminder -> {
-                String content = reminder.getContent();
-                long channelId = reminder.getChannelId();
-                Instant remindAt = reminder.getRemindAt();
+            if (totalPages > 1) {
+                remindersEmbed.setFooter("Page: %d/%d".formatted(pageToShow, totalPages));
 
-                remindersEmbed.addField(content, getDescription.apply(channelId, remindAt), false);
-            });
+                getPageEntries(pendingReminders, pageToShow)
+                    .forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+
+                listMessage.addActionRow(createPageTurnButtons(pageToShow, totalPages));
+            } else {
+                pendingReminders.forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+            }
         }
 
-        event.replyEmbeds(remindersEmbed.build()).setEphemeral(true).queue();
+        return listMessage.addEmbeds(remindersEmbed.build()).build();
     }
 
-    private Result<PendingRemindersRecord> getReminders(long guildId, long userId) {
-        return database.read(context -> context.selectFrom(PENDING_REMINDERS)
-            .where(PENDING_REMINDERS.GUILD_ID.eq(guildId)
-                .and(PENDING_REMINDERS.AUTHOR_ID.eq(userId)))
-            .orderBy(PENDING_REMINDERS.REMIND_AT.asc())
-            .fetch());
+    private List<Button> createPageTurnButtons(int pageNumber, int totalPages) {
```

Comment:
who is `pageNumber`? the current page? the page to turn to? ... please clarify

Reply:
currentPage, yes i also think it is a bit confusing

Reply:
done